apps/google-analytics-4/frontend/src/components/common/HyperLink/HyperLink.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { MouseEventHandler } from 'react';
+import { TextLink } from '@contentful/f36-components';
+
+interface Props {
+  body: string;
+  substring: string;
+  onClick?: MouseEventHandler<HTMLAnchorElement>;
+  hyperLinkHref?: string;
+}
+
+const HyperLink = (props: Props) => {
+  const { body, substring, onClick = () => {}, hyperLinkHref } = props;
+  const link = (
+    <TextLink onClick={onClick} href={hyperLinkHref} target="_blank" rel="noopener noreferer">
+      {substring}
+    </TextLink>
+  );
+
+  const formatLink = () => {
+    const bodyWithTextLink = body.split(substring).reduce((prev: any, current, i) => {
+      if (!i) {
+        return [current];
+      }
+      return prev.concat(link, current);
+    }, []);
+    return bodyWithTextLink as JSX.Element;
+  };
+
+  return formatLink();
+};
```

Comment:
Created this component, which can hopefully be used elsewhere, to easily "embed" a hyperlink (`TextLink` component) into a string dynamically. Ex: 
<img width="385" alt="Screenshot 2023-03-24 at 4 35 48 PM" src="https://user-images.githubusercontent.com/58186851/227984757-23943f4e-42ab-4e86-8840-3c61844fd4a3.png">